### PR TITLE
tag module

### PR DESCRIPTION
**Pull request by chkp-amiads**
##### SUMMARY

Checkpoint tag module

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

checkpoint
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/network/checkpoint/cp_mgmt_tag.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    - tag1
+    - tag2
+
+- name: set-tag
+  cp_mgmt_tag:
+    name: My New Tag1
+    new_name: My New Tag21
+    state: present
+    tags:
+      add: tag3
```
Comment (justjais) [inline lib/ansible/modules/network/checkpoint/cp_mgmt_tag.py]:
```suggestion
      add: tag3
```
`add` params need to be documented

**Review by justjais — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/checkpoint/cp_mgmt_tag.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+- name: add-tag
+  cp_mgmt_tag:
+    name: My New Tag1
+    state: present
+    tags:
+    - tag1
+    - tag2
+
+- name: set-tag
+  cp_mgmt_tag:
```
Comment (justjais) [inline lib/ansible/modules/network/checkpoint/cp_mgmt_tag.py]: @chkp-amiads can u update the set example here, as currently it's not doing anything extra what add example already performed

**Review by justjais — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/checkpoint/cp_mgmt_tag.py (hunk 2 above)
Comment (chkp-amiads) [reply lib/ansible/modules/network/checkpoint/cp_mgmt_tag.py]: Same as for dynaminc object and threat-layer, we can add example of generic attributes like comment, we would like to skip it as it is not a blocker

File: lib/ansible/modules/network/checkpoint/cp_mgmt_tag.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    name: My New Tag1
+    state: present
+    tags:
+    - tag1
+    - tag2
+
+- name: delete-tag
+  cp_mgmt_tag:
+    name: dda885bb-0d5b-4529-96eb-55b757847092
```
Comment (justjais) [inline lib/ansible/modules/network/checkpoint/cp_mgmt_tag.py]: @chkp-amiads here name should be `My New Tag1` correct?
```suggestion
    name: My New Tag1
```

**Review by justjais — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/checkpoint/cp_mgmt_tag.py (hunk 3 above)
Comment (chkp-amiads) [reply lib/ansible/modules/network/checkpoint/cp_mgmt_tag.py]: fixed

**Review by justjais — APPROVED** [review]

**Pull request merged**